client/src/controllers/ActionController.test.js

```diff
@@ -84,8 +84,61 @@ describe('ActionController', () => {
 
       expect(clickMock).toHaveBeenCalled();
     });
+
   });
 
+  describe('click method for textarea', () => {
+    beforeEach(async()=>{
+      await setup(`
+      <textarea 
+      id="text"
+      data-controller="w-action"
+      data-action="some-event->w-action#click" 
+      > 
+      Text </textarea> 
+      `)
+    })
+
+    it('should call click method then textarea is click via Stimulus action',()=>{
+      const text = document.getElementById('text');
+      const clickMock = jest.fn();
+      HTMLTextAreaElement.prototype.click = clickMock;
+
+      text.addEventListener('some-event', text.click());
+
+      const event = new CustomEvent('some-event');
+      text.dispatchEvent(event);
+
+      expect(clickMock).toHaveBeenCalled();
+    })
+  })
+
+  describe('click method for input', () => {
+    beforeEach(async()=>{
+      await setup(`
+      <input 
+      type="text"
+      id="input"
+      data-controller="w-action"
+      data-action="some-event->w-action#click" 
+      />
+      `)
+    })
+
+    it('should call click method then input is click via Stimulus action',()=>{
+      const input = document.getElementById('input');
+      const clickMock = jest.fn();
+      HTMLInputElement.prototype.click = clickMock;
+
+      input.addEventListener('some-event', input.click());
+
+      const event = new CustomEvent('some-event');
+      input.dispatchEvent(event);
+
+      expect(clickMock).toHaveBeenCalled();
+    })
+  })
+
```

Comment:
These tests are not exactly what is needed, this is testing the `click` method on the controller and not the `select` method.

If you look further down there is already a describe block for 'select', here's a proposed test suite that will cover the checking of the `input` element but it would be good to add a test that also ensures we do not get errors when calling `select` on a button element.


```js
describe('select method', () => {
    it('select should be called when you click on text in textarea', async () => {
      await setup(`
        <textarea
          id="text"
          rows="1"
          data-controller="w-action"
          data-action="focus->w-action#select"
        >
          some random text
        </textarea>
      `);

      const textarea = document.getElementById('text');

      // check that there is no selection initially
      expect(textarea.selectionStart).toBe(0);
      expect(textarea.selectionEnd).toBe(0);

      // focus
      textarea.focus();

      // check that there is a selection after focus
      expect(textarea.selectionStart).toBe(0);
      expect(textarea.selectionEnd).toBe(textarea.value.length);
    });

    it('select should be called for for input elements', async () => {
      await setup(`
      <input
        type="text"
        id="input"
        data-controller="w-action"
        data-action="some-event->w-action#select"
        value="some random text"
      />
      `);

      const input = document.getElementById('input');

      // check that there is no selection initially
      expect(input.selectionStart).toBe(0);
      expect(input.selectionEnd).toBe(0);

      const event = new CustomEvent('some-event');
      input.dispatchEvent(event);

      // check that there is a selection after the event
      expect(input.selectionStart).toBe(0);
      expect(input.selectionEnd).toBe(16);
    });
  });
```